content/academy/puppeteer_playwright/page/interacting_with_a_page.md

```diff
@@ -0,0 +1,170 @@
+
+```JavaScript
+// Type the query into the search box
+await page.type('input[title="Search"]', 'hello world');
+```
+
+Finally, we can press a single key by accessing the `keyboard` property of `page` and calling the `press()` function on it:
+
+```JavaScript
+// Press enter
+await page.keyboard.press('Enter');
+```
+
+So far, we've got this:
+
+```marked-tabs
+<marked-tab header="Playwright" lang="javascript">
+import { chromium } from 'playwright';
+
+const browser = await chromium.launch({ headless: false });
+
+const page = await browser.newPage();
+
+await page.goto('https://google.com/');
+
+// Click the "I agree" button
+await page.click('button:has-text("I agree")');
+
+// Type the query into the search box
+await page.type('input[title="Search"]', 'hello world');
+
+// Press enter
+await page.keyboard.press('Enter');
+
+await page.waitForTimeout(10000)
+await browser.close();
+</marked-tab>
+<marked-tab header="Puppeteer" lang="javascript">
+import puppeteer from 'puppeteer';
+
+const browser = await puppeteer.launch({ headless: false });
+
+const page = await browser.newPage();
+
+await page.goto('https://google.com/');
+
+// Click the "I agree" button
+await page.click('button + button:nth-child(2)');
+
+// Type the query into the search box
+await page.type('input[title="Search"]', 'hello world');
+
+// Press enter
+await page.keyboard.press('Enter');
+
+await page.waitForTimeout(10000)
+await browser.close();
+</marked-tab>
+```
+
+When we run it, we leave off on the results page:
+
+![Google results page reached by headless browser]({{@asset puppeteer_playwright/page/images/google-results.webp}})
+
+Great! So now all we have to do is click the first result, which matches the selector `.g a`:
+
+```marked-tabs
+<marked-tab header="Playwright" lang="javascript">
+import { chromium } from 'playwright';
+
+const browser = await chromium.launch({ headless: false });
+
+const page = await browser.newPage();
+
+await page.goto('https://google.com/');
+
+await page.click('button:has-text("I agree")');
+
+await page.type('input[title="Search"]', 'hello world');
+
+await page.keyboard.press('Enter');
+
+// Click the first result
+await page.click('.g a');
+
+await page.waitForTimeout(10000)
+await browser.close();
+</marked-tab>
+<marked-tab header="Puppeteer" lang="javascript">
+// This code will throw an error!
+import puppeteer from 'puppeteer';
+
+const browser = await puppeteer.launch({ headless: false });
+
+const page = await browser.newPage();
+
+await page.goto('https://google.com/');
+
+await page.click('button + button:nth-child(2)');
+
+await page.type('input[title="Search"]', 'hello world');
+
+await page.keyboard.press('Enter');
+
+// Click the first result
+await page.click('.g a');
+
+await page.waitForTimeout(10000)
+await browser.close();
+</marked-tab>
+```
+
+But wait, when we try to run the Puppeteer code, we run into this nasty error:
```

Comment:
I like that. Showing a problem and then continuing with a solution is a great teaching style.